x/tokenregistry/client/cli/query.go

```diff
@@ -168,9 +138,9 @@ func GetCmdGenerateEntry() *cobra.Command {
 			// generate denom if path is also provided.
 			// override the IBC generation with --denom if specified explicitly.
 			// otherwise fallback to base_denom
-			if ibcChannelId != "" {
+			if ibcChannelID != "" {
```

Comment:
This should proabably change to if ibcPath != "" and use the path instead of ibcChannelID.

If ibcChannelID is specified alone without ibcPath, then it should be used to set path and denom = ibc hash.

We should also introduce validation checks on RegistryEntry that are run here.